Ops dashboard trigger

IsaackMwenda opened the pull request:
1.Added a function that fetches messages, computes coding progress and pushes it to the metrics collection.
2.Added a line in add.py to call the function that fetches messages and computes coding progress

lukechurch commented on data_tools/compute_coding_progress.py:
```diff
@@ -6,36 +6,37 @@
 if (len(sys.argv) != 2):
     print ("Usage python compute_coding_progress.py coda_crypto_token")
-    exit(1)
+        exit(1)
```
Please remove additional indentation

lukechurch commented on data_tools/compute_coding_progress.py:
```diff
@@ -6,36 +6,37 @@
-
-ids = fcw.get_dataset_ids()
-
-data['coding_progress'] = {}
```
This change turns this code into a function - great - but it will break being able to run this tool from the command line. That's a bit of a problem.

Please can you add a __main__ method to the bottom of the file that calls the functions, so that the functions can be used either directly on the command line, or as functions

lukechurch replied:
Thanks for the change

lukechurch commented on data_tools/compute_coding_progress.py:
```diff
@@ -6,36 +6,37 @@
-        data['coding_progress'][id] = metrics   
-        continue
-        
+def compute_coding_progress(ids,messages,metrics):
```
spaces between variable names in the arg list please:

ids, messages, metrics

And please can you add a doc comment saying what they are? See some of the code in CoreDataModules for an example

lukechurch commented on data_tools/compute_coding_progress.py:
```diff
@@ -6,36 +6,37 @@
+        data['coding_progress'][id]['messages_count'] = len(messages)
+        data['coding_progress'][id]['messages_with_label'] = messages_with_labels
+    data["last_update"] = strftime("%Y-%m-%d %H:%M:%S", gmtime())
+    return json.dumps(data, indent=2)
```
This returns a string, it's better to have the code that computes the answer separate from the code that does something with that answer. 

So in this case, I think we should be returning data, and then the place that calls this function calling dumps

lukechurch commented on data_tools/compute_coding_progress.py:
```diff
@@ -6,36 +6,37 @@
+    return json.dumps(data, indent=2)
+    
+def get_messages_ids_metrics(): 
+    global ids = fcw.get_dataset_ids()
```
This isn't recommended as a way of passing arguments to a function - the global keyword should be used very sparingly. 

Instead please call call compute progress as:

``` progress = compute_coding_progress(ids, messages, metrics) ```

IsaackMwenda commented on data_tools/compute_coding_progress.py:
```diff
@@ -9,33 +9,55 @@
+    return progress
+
+if __name__ == "__main__":
+    get_messages_ids_metrics()
```
Hi @lukechurch  I don't understand this.

IsaackMwenda commented:
Hi @lukechurch  PTAL. Have;
Added a loop to fetch the metrics, messages for each dataset.
Removed unnecessary whitespaces at the end of the file.
Formatted function documentation.
